.circleci/config.yml

```diff
@@ -0,0 +1,198 @@
+          key: v1-gradle-{{ checksum "./android/build.gradle" }}-{{ checksum  "./android/app/build.gradle" }}-{{ arch }}
+      - run:
+          working_directory: ./android
+          name: Download Gradle dependencies
+          command: ./gradlew dependencies
+      - save_cache:
+          name: Save Gradle cache
+          paths:
+            - ~/.gradle
+          key: v1-gradle-{{ checksum "./android/build.gradle" }}-{{ checksum  "./android/app/build.gradle" }}-{{ arch }}
+
+  setup-flutter:
+    description: "setup Flutter"
+    steps:
+      - run:
+          name: Download flutter SDK
+          command: git clone -b stable https://github.com/flutter/flutter.git flutter
+      - run:
+          name: Setup environment variables
+          command: echo 'export PATH="$PATH:`pwd`/flutter/bin"'  >> $BASH_ENV
+      - run:
+          name: Flutter doctor
+          command: flutter doctor
+
+jobs:
+  lint:
+    description: "Run static analysis for Flutter"
+    docker:
+      - image: cirrusci/flutter:stable
+    steps:
+      - checkout
+      - build-flutter-project
+      - run:
+          name: Analyze code
+          command: flutter analyze lib
+
+  test:
+    description: "Run tests for Flutter"
+    docker:
+      - image: cirrusci/flutter:stable
+    steps:
+      - checkout
+      - build-flutter-project
+      - run:
+          name: Run tests
+          command: flutter test
+
+  android-integration-test-build:
+    description: "Build Android Flutter integration tests"
+    docker:
+      - image: cirrusci/flutter:stable
+    steps:
+      - checkout
+      - run:
+          name: Change api key
+          command: sed -i s/api_key/$API_KEY/ example/integration_test/app_test.dart
+      - run:
+          working_directory: example/android
+          command: |
+            # flutter build generates files in android/ for building the app
+            flutter build apk
+            ./gradlew app:assembleAndroidTest
+            ./gradlew app:assembleDebug -Ptarget=`pwd`/../integration_test/app_test.dart
+      - persist_to_workspace:
+          root: .
+          paths:
+            - example/build/app/outputs/apk/debug/app-debug.apk
+            - example/build/app/outputs/apk/androidTest/debug/app-debug-androidTest.apk
+
+  run-firebase-tests:
+    description: "Run integration tests for Flutter"
+    executor: gcp-cli/google
+    steps:
+      - checkout
+      - attach_workspace:
+          at: .
+      - gcp-cli/initialize:
+          gcloud-service-key: GCLOUD_SERVICE_KEY
+          google-compute-zone: GOOGLE_COMPUTE_ZONE
+          google-project-id: GOOGLE_PROJECT_ID
+      - run:
+          name: Test with Firebase Test Lab
+          command: >
+            gcloud firebase test android run --type instrumentation \
+              --app example/build/app/outputs/apk/debug/app-debug.apk \
+              --test example/build/app/outputs/apk/androidTest/debug/app-debug-androidTest.apk \
+              --timeout 2m \
```

Comment:
I have the same concerns about reusing the example app for integration tests as I mentioned in the cordova PR, they will evolve in different ways with different responsibilities